doc/api-extensions.md

```diff
@@ -2388,3 +2388,8 @@ For more information on access control for OIDC clients, see {ref}`fine-grained-
 ## `vm_disk_io_limits`
 
 Adds the ability to limit disk I/O for virtual machines.
+
+## `ct_linux_kernel_modules_dynamic_load`
```

Comment:
Lets call this `container_syscall_intercept_finit_modules` to align with other intercepted syscalls extensions.

Reply:
Also please can this commit come first in the PR to aid with backporting - allows me to see what the following commits are about.

Reply:
Or should it be `finit_module_parse`?

Reply:
While the `finit_module` syscall designation is technically correct, I had never hear of it before. If we want self descriptive API extension names, maybe one of those would be more friendly to non-kernel devs?

* `container_linux_kernel_modules_load_ondemand`
* `container_linux_kernel_modules_ondemand_loading`

In any case, I think we should have either `dynamic` or `ondemand` (whichever we settle on) in the name to make it easier to link with the actual configuration key/value needed.

Reply:
I think it needs to have interception in it at least so it goes along with the other ones and is clear what is happening.

Reply:
Have fixed!